### ISSUE TITLE
Develop branch failed becauce paddle::framework::details::OpHandleBase::dev_ctxes_ is protected

### ISSUE THREAD

**Issue by Xreki** [PaddlePaddle/Paddle#10085]
The detailed error is:
```
In file included from /paddle/paddle/fluid/framework/details/reduce_op_handle.h:21:0,
                 from /paddle/paddle/fluid/framework/details/reduce_op_handle_test.cc:15:
/paddle/paddle/fluid/framework/details/op_handle_base.h: In member function 'void paddle::framework::details::TestReduceOpHandle::InitReduceOp(size_t)':
/paddle/paddle/fluid/framework/details/op_handle_base.h:75:7: error: 'std::unordered_map<boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace>, paddle::platform::DeviceContext*, paddle::platform::PlaceHash> paddle::framework::details::OpHandleBase::dev_ctxes_' is protected
       dev_ctxes_;
       ^
/paddle/paddle/fluid/framework/details/reduce_op_handle_test.cc:112:21: error: within this context
         op_handle_->dev_ctxes_[gpu_list_[j]] = ctxs_[j].get();
                     ^
paddle/fluid/framework/details/CMakeFiles/reduce_op_handle_test.dir/build.make:62: recipe for target 'paddle/fluid/framework/details/CMakeFiles/reduce_op_handle_test.dir/reduce_op_handle_test.cc.o' failed
make[2]: *** [paddle/fluid/framework/details/CMakeFiles/reduce_op_handle_test.dir/reduce_op_handle_test.cc.o] Error 1
CMakeFiles/Makefile2:14670: recipe for target 'paddle/fluid/framework/details/CMakeFiles/reduce_op_handle_test.dir/all' failed
make[1]: *** [paddle/fluid/framework/details/CMakeFiles/reduce_op_handle_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by luotao1** [comment]
+1 same error to me

**Comment by typhoonzero** [comment]
@reyoung @chengduoZH Can you please take a look at this?

**Comment by Xreki** [comment]
It is fixed in #10079
